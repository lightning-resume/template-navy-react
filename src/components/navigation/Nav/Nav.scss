@import 'src/styles/colors.scss';

.nav-container{
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: inherit;
  color: $lightest-slate;
}

.nav {
  display: flex;
  gap: 20;
}

.nav-logo {
  font-size: 1rem;
    font-weight: bold;
    box-sizing: border-box;
    padding: 10px;
    border: dotted;
    border-color: $green;
}

.nav-link {
  box-sizing: border-box;
  margin: 0 5px;
  text-decoration: none;
  font-size: 1rem;
  font-weight: bold;
  border-bottom: none;
  color: $lightest-slate;
  transition: color 0.5s ease-in-out;
  padding-bottom: 10px;

  &:hover {
    color: $green;
  }
  &.active {
    color: $green;
    border-bottom: solid 2px $green;
  }

  .text-selected {
    border-bottom: solid;
    color: $green;
  }
}