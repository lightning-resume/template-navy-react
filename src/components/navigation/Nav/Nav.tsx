import { useState, MouseEvent, useCallback, Fragment } from 'react'
import { capitalizeFirstLetter } from 'src/utils/capitalizeLetter'
import { useDisableScroll } from 'src/utils/hooks/useDisableScrool'
import { BurgerMenu } from '../BurgerMenu'

import emailSVG from 'src/assets/email.svg'
import linkedinSVG from 'src/assets/linkedin.svg'

import './Nav.scss'

const Links = ({
  links,
  hash,
  handleActive,
}: {
  links: string[]
  hash: string
  handleActive: (event: MouseEvent) => void
}) => {
  return (
    <Fragment>
      {links.map(link => (
        <a
          key={link}
          className={['nav-link', hash.includes(link) && 'active'].filter(ele => ele).join(' ')}
          href={`#${link}`}
        >
          <span title={link} onClick={handleActive}>
            {capitalizeFirstLetter(link)}
          </span>
        </a>
      ))}
    </Fragment>
  )
}
interface NavProps {
  onClick?: () => {}
  links: string[]
  profile: Resume.Profile
}

const DEVICE_DESKTOP = 'desktop'

const isShown = (show: string) => show === 'show'

export const Nav = ({ links, profile }: NavProps) => {
  const [hash, setHash] = useState(window?.location?.hash || links[0])
  const [show, setShow] = useState('hide')
  const handleShowBanner = useCallback(() => {
    return setShow(isShown(show) ? 'hide' : 'show')
  }, [show])
  const handleActive = useCallback(
    (event: MouseEvent, device?: string) => {
      const target = event.target as HTMLElement
      setHash(target.title)
      if (device === DEVICE_DESKTOP) return
      handleShowBanner()
    },
    [hash, show],
  )
  useDisableScroll([isShown(show)])
  return (
    <Fragment>
      <div className="nav-container">
        <div className="nav-logo-container">
          <div className="nav-logo">{capitalizeFirstLetter(profile.name)[0]}</div>
          <div className="nav-logo-socials">
            {profile.email && (
              <a href={`mailto:${profile.email}`}>
                <img src={emailSVG} height="15px" width="15px" />
              </a>
            )}
            {profile.linkedin && (
              <a href={`https://${profile.linkedin}`} target={'_blank'} rel="noreferrer">
                <img src={linkedinSVG} height="15px" width="15px" />
              </a>
            )}
          </div>
        </div>
        <div className="nav-burger">
          <BurgerMenu isActive={isShown(show)} onClick={handleShowBanner} />
        </div>
        <nav className="nav">
          <Links links={links} hash={hash} handleActive={ev => handleActive(ev, DEVICE_DESKTOP)} />
        </nav>
      </div>
      <div className="nav-banner-container">
        <div
          className={`nav-banner-lightbox ${show}`}
          onClick={isShown(show) ? handleShowBanner : () => {}}
        />
        <div className={`nav-banner ${show}`}>
          <div className={'nav-banner-content'}>
            <Links links={links} hash={hash} handleActive={handleActive} />
          </div>
        </div>
      </div>
    </Fragment>
  )
}
