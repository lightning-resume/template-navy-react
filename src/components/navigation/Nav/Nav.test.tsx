import { screen, render, fireEvent } from '@testing-library/react'

import { Nav } from './Nav'

const mockProfile = {
  name: 'test',
  location: 'Madrid',
  email: '',
  linkedin: '',
}

const setMockProps = (props?: { links?: string[]; profile?: Resume.Profile }) => {
  return {
    links: ['first', 'second'],
    profile: mockProfile,
    ...props,
  }
}

describe('Nav', () => {
  describe('social links', () => {
    it('should render email if exist', () => {
      render(<Nav {...setMockProps({ profile: { ...mockProfile, email: 'test@asd.com' } })} />)
      const node = screen.getByLabelText('mail to')
      expect(node).toBeTruthy()
      expect(node).toMatchSnapshot()
    })
    it('should no render email if no exist', () => {
      render(<Nav {...setMockProps()} />)

      expect(screen.queryByLabelText('mail to')).toBeFalsy()
    })
    it('should render linkedin if exist', () => {
      render(<Nav {...setMockProps({ profile: { ...mockProfile, linkedin: 'test@linkedin.com' } })} />)

      const node = screen.getByLabelText('linkedin profile')
      expect(node).toBeTruthy()
      expect(node).toMatchSnapshot()
    })
    it('should no render linkedin if no exist', () => {
      render(<Nav {...setMockProps()} />)

      expect(screen.queryByLabelText('linkedin profile')).toBeFalsy()
    })
    it('should render the burger menu', () => {
      render(<Nav {...setMockProps()} />)

      expect(screen.queryByLabelText('navigation menu')).toBeTruthy()
    })
    it('should render lightbox as hidden', () => {
      render(<Nav {...setMockProps()} />)

      expect(screen.queryByLabelText('navigation-lightbox-hide')).toBeTruthy()
      expect(screen.queryByLabelText('navigation-lightbox-show')).toBeFalsy()
    })
  })
  describe('onClick burgerMenu', () => {
    it('should change shown state of lightbox', () => {
      render(<Nav {...setMockProps()} />)

      expect(screen.queryByLabelText('navigation-lightbox-hide')).toBeTruthy()

      fireEvent.click(screen.queryByLabelText('navigation menu') as HTMLElement)

      expect(screen.queryByLabelText('navigation-lightbox-show')).toBeTruthy()
    })
    it('should change shown state of banner', () => {
      render(<Nav {...setMockProps()} />)

      expect(screen.queryByLabelText('navigation-banner-hide')).toBeTruthy()

      fireEvent.click(screen.queryByLabelText('navigation menu') as HTMLElement)

      expect(screen.queryByLabelText('navigation-banner-show')).toBeTruthy()
    })
  })
  describe('onClick lightbox', () => {
    it('should change shown state of lightbox', () => {
      render(<Nav {...setMockProps()} />)

      expect(screen.queryByLabelText('navigation-lightbox-hide')).toBeTruthy()

      fireEvent.click(screen.queryByLabelText('navigation menu') as HTMLElement)

      expect(screen.queryByLabelText('navigation-lightbox-show')).toBeTruthy()

      fireEvent.click(screen.queryByLabelText('navigation-lightbox-show') as HTMLElement)

      expect(screen.queryByLabelText('navigation-lightbox-hide')).toBeTruthy()
    })
    it('should change shown state of banner', () => {
      render(<Nav {...setMockProps()} />)

      expect(screen.queryByLabelText('navigation-banner-hide')).toBeTruthy()

      fireEvent.click(screen.queryByLabelText('navigation menu') as HTMLElement)

      expect(screen.queryByLabelText('navigation-banner-show')).toBeTruthy()

      fireEvent.click(screen.queryByLabelText('navigation-lightbox-show') as HTMLElement)

      expect(screen.queryByLabelText('navigation-banner-hide')).toBeTruthy()
    })
  })
})
