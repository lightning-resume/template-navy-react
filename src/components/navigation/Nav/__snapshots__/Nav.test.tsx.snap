// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav social links should render email if exist 1`] = `
<a
  aria-label="mail to"
  href="mailto:test@asd.com"
>
  <img
    height="15px"
    src="email.svg"
    width="15px"
  />
</a>
`;

exports[`Nav social links should render linkedin if exist 1`] = `
<a
  aria-label="linkedin profile"
  href="https://test@linkedin.com"
  rel="noreferrer"
  target="_blank"
>
  <img
    height="15px"
    src="linkedin.svg"
    width="15px"
  />
</a>
`;
